### PR TITLE
Troops comp + service

### PR DESCRIPTION
**Pull request by lukasrehor**

### CODE REVIEW

File: src/app/_models/troop.model.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+export class Troop {
+  id: number;
+  level: number;
+  hp: number;
+  attack: number;
+  defence: number;
+  started_at: number;
```
Comment (balazsbarni) [inline src/app/_models/troop.model.ts]: `startedAt` 
`https://github.com/airbnb/javascript#naming-conventions
`

File: src/app/game/troops/troop/troop.component.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
```
Comment (balazsbarni) [inline src/app/game/troops/troop/troop.component.ts]: what is this component does?

File: src/app/game/troops/troops.service.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class TroopsService {
+
+  constructor(private http: HttpClient) {}
+
+  run() {
```
Comment (balazsbarni) [inline src/app/game/troops/troops.service.ts]: `run`? maybe a more descriptive function name

File: src/app/game/troops/troops.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+describe('TroopsComponent', () => {
+  let component: TroopsComponent;
+  let fixture: ComponentFixture<TroopsComponent>;
+  let service: TroopsService;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [TroopsComponent],
+      imports: [HttpClientModule],
+      providers: [TroopsService]
```
Comment (balazsbarni) [inline src/app/game/troops/troops.component.spec.ts]: maybe mock the `TroopService`, not provide it

File: src/app/_models/troop.model.ts (hunk 1 above)
Comment (lukasrehor) [reply src/app/_models/troop.model.ts]: changed

File: src/app/game/troops/troop/troop.component.ts (hunk 2 above)
Comment (lukasrehor) [reply src/app/game/troops/troop/troop.component.ts]: deleted

File: src/app/game/troops/troops.service.ts (hunk 3 above)
Comment (lukasrehor) [reply src/app/game/troops/troops.service.ts]: renamed

File: src/app/game/troops/troops.component.spec.ts (hunk 4 above)
Comment (lukasrehor) [reply src/app/game/troops/troops.component.spec.ts]: changed to mock the service

File: src/app/_helpers/mock-troops.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import { Troop } from '../_models/troop.model';
+
+export const Troops: Troop[] = [
```
Comment (balazsbarni) [inline src/app/_helpers/mock-troops.ts]: `troops`

File: src/app/game/troops/troops.component.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  ngOnInit() {
+    let values =  this.troopsService.getTroopsAndStats();
+    this.levels = values.levels;
+    this.totalAttack = values.totalAttack;
+    this.totalDefence = values.totalDefence
+    this.sustenance = values.sustenance;
+  }
+
+  checkLevels() {
+    let keys = Object.keys(this.levels);
```
Comment (balazsbarni) [inline src/app/game/troops/troops.component.ts]: `const`

File: src/app/game/troops/troops.service.ts [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+      sustenance = troops.length;
+    });
+    return {levels,totalAttack,totalDefence,sustenance};
+  }
+
+  getTroops(): Observable<any> {
+    return this.http.get('/game/troops', {observe: 'response'});
+  }
+
+  setLevels (troops): any {
```
Comment (balazsbarni) [inline src/app/game/troops/troops.service.ts]: `calculateTroopLevels` maybe?

File: src/app/_helpers/mock-troops.ts (hunk 5 above)
Comment (lukasrehor) [reply src/app/_helpers/mock-troops.ts]: done

File: src/app/game/troops/troops.component.ts (hunk 6 above)
Comment (lukasrehor) [reply src/app/game/troops/troops.component.ts]: changed

File: src/app/game/troops/troops.service.ts (hunk 7 above)
Comment (lukasrehor) [reply src/app/game/troops/troops.service.ts]: why not. done

**Review by balazsbarni — APPROVED** [review]

**Pull request merged**